streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -116,19 +138,33 @@ private void determineIsQueryable(final String queryableStoreName) {
     private <T> KTable<K, T> doAggregate(final ProcessorSupplier<K, Change<V>> aggregateSupplier,
                                          final String functionName,
                                          final StateStoreSupplier<KeyValueStore> storeSupplier) {
-        String sinkName = builder.newName(KStreamImpl.SINK_NAME);
-        String sourceName = builder.newName(KStreamImpl.SOURCE_NAME);
-        String funcName = builder.newName(functionName);
+        final String sinkName = builder.newName(KStreamImpl.SINK_NAME);
+        final String sourceName = builder.newName(KStreamImpl.SOURCE_NAME);
+        final String funcName = builder.newName(functionName);
+
+        buildAggregate(aggregateSupplier,
+                       storeSupplier.name() + KStreamImpl.REPARTITION_TOPIC_SUFFIX,
+                       funcName,
+                       sourceName,
+                       sinkName);
+        builder.internalTopologyBuilder.addStateStore(storeSupplier, funcName);
```

Comment:
`addStateStore` logic can also be in the `buildAggregate` function by adding the `storeSupplier` along with `aggregateSupplier`?

Reply:
No it can't as `buildAggregate` is also called from the other `doAggregate` that builds a `StateStoreBuilder`, so they need to be separate calls.